src/Symfony/Component/Asset/Package.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset;
+
+use Symfony\Component\Asset\Context\ContextInterface;
+use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+/**
+ * Basic package that adds a version to asset URLs.
+ *
+ * @author Kris Wallsmith <kris@symfony.com>
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class Package implements PackageInterface
+{
+    private $versionStrategy;
+    private $context;
+
+    public function __construct(VersionStrategyInterface $versionStrategy)
+    {
+        $this->versionStrategy = $versionStrategy;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getVersion($path)
+    {
+        return $this->versionStrategy->getVersion($path);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUrl($path)
+    {
+        if ($this->isAbsoluteUrl($path)) {
+            return $path;
+        }
+
+        return $this->versionStrategy->applyVersion($path);
+    }
+
+    public function setContext(ContextInterface $context)
```

Comment:
what do you think about making the context a mandatory argument (or an optional argument in the constructor but building an instance internally in this case) instead of an optional deps, and providing a basic context using static values passed in its constructor (with the default values of these arguments matching the current behavior where there is no context) ? It would simplify the code by avoiding to check for null everywhere (null-object pattern)